File: climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int a = 1, b = 2, c = 3;
        if (n == 1) {
            return 1;
        } else if (n == 2) {
            return 2;
        }
        for (int i = 0; i < n-3; i++) {
            a = b, b = c;
            c = a + b;
        }
        return c;
    }
};

File: climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        a, b, c = 1, 2, 3
        if n == 1:
            return 1
        elif n == 2:
            return 2
        for i in range(n-3):
            a, b = b, c
            c = a + b
        return c
